website_updater.py

```diff
@@ -13,79 +13,94 @@
-        return response.read().decode('utf-8')
+        return response.read().decode("utf-8")
 
 
-def execute_cmd(cmd, cwd):
-    logging.info("Executing %s in %s", cmd, cwd)
-    subprocess.check_call(cmd, cwd=cwd, stdout=subprocess.DEVNULL)
+def execute_cmd(cmd: List[str], cwd: str, dry_run: bool):
+    log.info("Executing %s in %s", cmd, cwd)
+    if dry_run:
+        log.info("Dry run enabled; doing nothing")
+    else:
+        subprocess.check_call(cmd, cwd=cwd, stdout=subprocess.DEVNULL)
 
 
 def main():
     logging.basicConfig(level=logging.INFO)
 
     parser = argparse.ArgumentParser(
-        prog='website_updater',
-        description='Automatically updates the chombo.club website using CSV data')
-    parser.add_argument('--data', required=True, help='CSV data URL')
-    parser.add_argument('--template', required=True, help='Template directory path')
-    parser.add_argument('--repository', required=True, help='git repository path')
+        prog="website_updater",
+        description="Automatically updates the chombo.club website using CSV data",
+    )
+    parser.add_argument("--data", required=True, help="CSV data URL")
+    parser.add_argument("--template", required=True, help="Template directory path")
+    parser.add_argument("--repository", required=True, help="git repository path")
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Only print the changes and don't modify any files",
+    )
     args = parser.parse_args()
 
-    process(args.data, args.template, args.repository)
+    process(args.data, args.template, args.repository, args.dry_run)
 
 
-def process(data_url: str, template_dir: str, repo: str):
+def process(data_url: str, template_dir: str, repo: str, dry_run: bool):
     responses = request_responses(data_url)
+    num_countries = calc_countries(responses)
+    num_mers = calc_mers(responses)
 
-    env = Environment(
-        loader=FileSystemLoader("."),
-        autoescape=select_autoescape()
-    )
+    env = Environment(loader=FileSystemLoader("."), autoescape=select_autoescape())
 
-    now = datetime.datetime.now(pytz.timezone('Europe/Warsaw'))
+    now = datetime.datetime.now(pytz.timezone("Europe/Warsaw"))
     datetime_str = now.strftime("%Y%m%d_%H%M%S")
     branch_name = f"update_{datetime_str}"
 
-    cleanup_repo(repo, branch_name)
+    cleanup_repo(repo, branch_name, dry_run)
 
     path = pathlib.Path(template_dir)
-    for file in path.rglob('*.tmpl'):
+    for file in path.rglob("*.tmpl"):
         template = env.get_template(str(file))
 
         relative_path = file.relative_to(path)
         new_path = pathlib.Path(repo) / relative_path
-        logging.info("Writing to: %s", new_path)
+        log.info("Writing to: %s", new_path)
 
-        with open(str(new_path)[:-len(".tmpl")], 'w') as f:
-            f.write(template.render(data=responses, now=now))
+        with open(str(new_path)[: -len(".tmpl")], "w") as f:
```

Comment:
wdyt: (added `resolve` to straighten out the path after the relative path magic above) (man I love pathlib)
```suggestion
        new_path = new_path.resolve()
        with (new_path.parent / new_path.stem).open("w") as f:
```